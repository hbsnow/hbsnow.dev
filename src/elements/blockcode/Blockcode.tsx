import React from 'react'
import SyntaxHighlighter from 'react-syntax-highlighter'
import Pre from './Pre'
import Code from './Code'

const Blockcode: React.FC<BlockcodeProps> = ({
  children,
  language,
  ...restProps
}) => {
  return (
    <div data-testid="Blockcode" className="blockcode" {...restProps}>
      {language && <div className="language">{language}</div>}
      <SyntaxHighlighter
        language={language}
        style={false}
        PreTag={Pre}
        CodeTag={Code}
      >
        {children}
      </SyntaxHighlighter>
      <style jsx>{`
        .blockcode {
          background-color: var(--color-default-surface);
          margin-left: calc(var(--gap-size) * -2);
          margin-right: calc(var(--gap-size) * -2);
          margin-bottom: calc(var(--gap-size) * 2);
          padding: calc(var(--gap-size) * 1.5) calc(var(--gap-size) * 2);
          ${language && `padding-top: 0; `}
        }

        .language {
          display: inline-block;
          vertical-align: middle;
          background-color: rgba(0, 0, 0, 0.05);
          font-family: var(--font-family-code);
          font-size: 0.75rem;
          margin-bottom: var(--gap-size);
          padding: 0.125rem 0.5rem;
          transform: translateX(-0.5rem);
        }
      `}</style>
    </div>
  )
}

type BlockcodeProps = {
  language?: string
} & JSX.IntrinsicElements['div']

export default Blockcode
