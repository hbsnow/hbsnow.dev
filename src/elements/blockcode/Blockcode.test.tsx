/* eslint-env jest */
import React from 'react'
import { render, screen } from '@testing-library/react'
import Blockcode from './Blockcode'

describe('Blockcode Component', () => {
  it('render the blockcode', () => {
    render(<Blockcode>blockcode</Blockcode>)
    const target = screen.getByTestId('Blockcode')
    expect(target).toHaveTextContent('blockcode')
  })
})
