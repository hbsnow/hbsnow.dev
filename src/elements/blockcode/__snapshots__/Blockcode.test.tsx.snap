// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockcode component with language 1`] = `
<div
  className="jsx-2749244367 blockcode"
  data-testid="Blockcode"
>
  <div
    className="jsx-2749244367 language"
  >
    test
  </div>
  <pre
    className="jsx-1898440872 jsx-2149608739 hljs"
    data-testid="Pre"
  >
    <code
      className="jsx-4275953942 code"
      data-testid="Code"
    >
      <span
        className="hljs-attribute"
        style={Object {}}
      >
        blockcode
      </span>
    </code>
  </pre>
</div>
`;

exports[`Blockcode components that do not have a language 1`] = `
<div
  className="jsx-902339241 blockcode"
  data-testid="Blockcode"
>
  <pre
    className="jsx-1898440872 jsx-2149608739 hljs"
    data-testid="Pre"
  >
    <code
      className="jsx-4275953942 code"
      data-testid="Code"
    >
      blockcode
    </code>
  </pre>
</div>
`;
